be/src/util/jni_helper.h

```diff
@@ -0,0 +1,82 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Copied from
+//   https://github.com/apache/hadoop/blob/release-2.7.1/hadoop-hdfs-project/hadoop-hdfs/src/main/native/libhdfs/jni_helper.h
+// We need Hadoop 2.7 for Hortonworks HDP support, but libhdfs no longer exports
+// getJNIEnv so we have to copy some source files like this to Impala. ROCANA-3465
+
+#ifndef LIBHDFS_JNI_HELPER_H
+#define LIBHDFS_JNI_HELPER_H
```

Comment:
Good question. I originally wanted to include the actual hdfs header files instead of writing our own, but I couldn't find them. `thirdparty/hadoop-*/include` doesn't have very many files in it. So as far as I know we can't accidentally include the actual header from libhdfs.

Performing a `grep -r getJNIEnv .` in `thirdparty/hadoop-2.7.1.2.3.0.0-2557` only finds matching binary files (libhdfs.a and libhdfs.so.0.0.0).